File: 1436-destination-city.ts
// https://leetcode.com/problems/destination-city/
function destCity(paths: string[][]): string {
  const hasOutgoing = new Set();

  for (let i = 0; i < paths.length; i++) {
    hasOutgoing.add(paths[i][0]);
  }

  for (let i = 0; i < paths.length; i++) {
    if (!hasOutgoing.has(paths[i][1])) return paths[i][1];
  }
}

File: 1436-destination-city.py
#https://leetcode.com/problems/destination-city/
class Solution:
    def destCity(self, paths: List[List[str]]) -> str:
        has_outgoing = set()

        for path in paths:
            has_outgoing.add(path[0])

        for path in paths:
            if not path[1] in has_outgoing:
                return path[1]